### DISCUSSION TITLE
gf.Component.each_overlapping_inst()

### DISCUSSION THREAD

**Discussion by emptylambda** [gdsfactory/gdsfactory#2891] [General]
Hello all, 

First thanks for great work, wanting to ask about the `each_overlapping_inst(b)` function as being documented [here](https://gdsfactory.github.io/gdsfactory/_autosummary/gdsfactory.Component.html#gdsfactory.Component)
This method seems to be not implemented?

**Comment by sebastian-goeldi** [comment]
What makes you say that it's not been implemented? This is a (almost) transparently proxied function from `klayout.db`:

<details open>

<summary>test.py</summary>


```python
import gdsfactory as gf

c = gf.Component()

b1 = c << gf.components.straight()
b2 = c << gf.components.bend_circular()

b2.connect("o1", b1, "o2", mirror=True)

c.show()

for inst in c.each_overlapping_inst(b2.bbox().enlarged(-1)):
    print(f"found {inst.cell.name} at {inst.dcplx_trans}")
```

</details>

output:
```bash
python test.py 
found bend_circular_RNone_A90_35590aa7 at m0 *1 10,0
2024-07-09 02:07:38.777 | INFO     | kfactory.kcell:show:7851 - klive v0.3.3: Reloaded file '/home/sgoeldi/repos/gdsfactory/build/gds/test.oas'
```
